### DISCUSSION TITLE
spreadsheetimporter vs document.onpaste

### DISCUSSION THREAD

**Discussion by octavonu** [marianfoo/ui5-cc-excelUpload#599] [General]
Hi,

What would be the benefit of using spreadsheetimporter vs standard document.onpaste event? 
https://developer.mozilla.org/en-US/docs/Web/API/Document/paste_event

I am asking this because I am working on a small project where user must upload data from excel into backend. Using the browser standard  onpaste event i can read the pasted values and display them in a ui5 table control and after do whatever i want with them. 

Thank you

**Comment by marianfoo** [comment]
Hi,
You are right, this method also exists and is also used in standard Fiori Elements applications.https://community.sap.com/t5/application-development-blog-posts/streamlining-data-entry-exploring-the-new-paste-feature-in-fiori-elements/ba-p/13710973
There is no reason not to use it.
However, Spreadsheetimporter does a lot that you would have to develop additionally.
It takes the Excel, extracts the data, parses the data for the OData V2 or OData V4 format and sends the data to the backend.
It also provides a control interface, error handling and extension points.
All things you would have to develop yourself.
But if it works for you in your case, there is no reason not to keep it that way.

**Comment by octavonu** [comment]
Hi @marianfoo 

Thank you for your input.

Indeed, it makes sense to use Spreadsheetimporter in OData context.

Have a nice day
